nixos/modules/services/hardware/intel-lpmd.nix

```
@@ -0,0 +1,527 @@
+        '';
+
+        type = with lib.types; bool;
+      };
+
+      util_entry_threshold = lib.mkOption {
+        default = 10;
+        description = ''
+          Specifices the system utilization threshold (in %) for entering LPM.
+          Set both this and `util_exit_threshold` to '0' to disable this feature.
+        '';
+
+        type = with lib.types; numbers.between 0 100;
+      };
+
+      util_exit_threshold = lib.mkOption {
+        default = 95;
+        description = ''
+          Specifices the system utilization threshold (in %) for exiting LPM.
+          Set both this and `util_entry_threshold` to '0' to disable this feature.
+        '';
+
+        type = with lib.types; numbers.between 0 100;
+      };
+
+      EntryDelayMS = lib.mkOption {
+        default = 0;
+        description = ''
+          Specifies the sample interval (in ms) used by the utilization monitor
+          when system wants to enter LPM.
+
+          Setting this to '0' uses the default interval of 1000ms.
+        '';
+
+        type = with lib.types; ints.unsigned;
+      };
+
+      ExitDelayMS = lib.mkOption {
+        default = 0;
+        description = ''
+          Specifies the sample interval (in ms) used by the utilization monitor
+          when system wants to exit LPM.
+
+          Setting this to '0' uses an adaptive value which is calculated based
+          on CPU utilization.
+        '';
+
+        type = with lib.types; ints.unsigned;
+      };
+
+      # TODO: it is unclear if disabling requires both to be set to '0' or not.
+      EntryHystMS = lib.mkOption {
+        default = "0";
+        description = ''
+          Specifies a hysteresis threshold (in ms) when the system is in LPM.
+          When the previous average time **stayed in** LPM is lower than this
+          value, the **current enter** LPM request will be ignored, due to the
+          expectation of the system exiting LPM soon.
+
+          Setting this to '0' disables the hysteresis algorithm.
+        '';
+
+        type = with lib.types; ints.unsigned;
+      };
+
+      ExitHystMS = lib.mkOption {
+        default = "0";
+        description = ''
+          Specifies a hysteresis threshold (in ms) when the system is not in LPM.
+          When the previous average time **stayed out of** LPM is lower than this
+          value, the **current exit** LPM request will be ignored, due to the
+          expectation of the system entering LPM soon.
+
+          Setting this to '0' disables the hysteresis algorithm.
+        '';
+
+        type = with lib.types; ints.unsigned;
+      };
+
+      lp_mode_epp = lib.mkOption {
+        default = null;
+        description = ''
+          EPP (energy performance preferences) to use in LPM.
+          Leave as `null` to ignore this setting.
+
+          Accepts a value from **0** (favor performance)
+          to **255** (favor power).
+        '';
+
+        type = with lib.types; nullOr numbers.between 0 255;
```

```suggestion
      lp_mode_epp = lib.mkOption {
        type = with lib.types; nullOr numbers.between 0 255;
        default = null;
        description = ''
          EPP (energy performance preferences) to use in LPM.
          Leave as `null` to ignore this setting.

          Accepts a value from **0** (favor performance)
          to **255** (favor power).
        '';
```

We should put the 0 to 255 range into the type.